editor/plugins/theme_editor_plugin.h

```diff
@@ -41,6 +41,87 @@
 
 #include "editor/editor_node.h"
 
+class ThemeItemEditorDialog : public AcceptDialog {
+	GDCLASS(ThemeItemEditorDialog, AcceptDialog);
+
+	Ref<Theme> edited_theme;
+
+	ItemList *edit_type_list;
+	OptionButton *edit_add_class_options;
+	LineEdit *edit_add_custom_value;
+	String edited_item_type;
+
+	enum DataType {
+		DATA_TYPE_ICON,
+		DATA_TYPE_STYLEBOX,
+		DATA_TYPE_FONT,
+		DATA_TYPE_FONT_SIZE,
+		DATA_TYPE_COLOR,
+		DATA_TYPE_CONSTANT,
+	};
+
+	Button *edit_items_add_color;
+	Button *edit_items_add_constant;
+	Button *edit_items_add_font;
+	Button *edit_items_add_font_size;
+	Button *edit_items_add_icon;
+	Button *edit_items_add_stylebox;
+	Button *edit_items_remove_class;
+	Button *edit_items_remove_custom;
+	Button *edit_items_remove_all;
+	Tree *edit_items_tree;
+
+	enum ItemsTreeAction {
+		ITEMS_TREE_RENAME_ITEM,
+		ITEMS_TREE_REMOVE_ITEM,
+		ITEMS_TREE_REMOVE_DATA_TYPE,
+	};
+
+	ConfirmationDialog *edit_theme_item_dialog;
+	VBoxContainer *edit_theme_item_old_vb;
+	Label *theme_item_old_name;
+	LineEdit *theme_item_name;
+
+	enum ItemPopupMode {
+		CREATE_THEME_ITEM,
+		RENAME_THEME_ITEM,
+	};
+
+	int item_popup_mode;
+	String edit_item_old_name;
+	int edit_item_data_type;
```

Comment:
You can initialize the `int` to `0` here (not needed for `String` as it's default initialized).

That being said aren't those two `int`s meant to be enum types instead?

Reply:
> You can initialize the int to 0 here (not needed for String as it's default initialized).

Oh, okay. I wasn't really sure if it's better to initialize them in the header or in the `.cpp`, as I've seen both in our codebase. Here in particular `time_left` was initialized inside of the constructor, so I went the same way with my code.

Same for `int`s vs `enum`s. `popup_mode` from the previous version was stored as an `int`, so I did the same.

Reply:
Yeah prior to 4.0 we initialized in the constructor as that's the only option pre-C++11. With 4.0 we're finally porting the codebase to C++11 (actually C++17) so we can now benefit from member initialization directly in the header.

And for enums it's usually preferred to use enum types as that gives you compile time checks that the values you're passing are valid. The only issue is that when passing values from GDScript it's always `int` so it needs to be validated "manually", but here for editor code that shouldn't be the case.